packages/webapp/src/containers/Task/saga.js

```diff
@@ -533,10 +534,32 @@ export function* createTaskSaga({ payload }) {
       header,
     );
     if (result) {
+      const task = data.task_type_id == 8 ? result.data[0] : result.data;
+
+      const task_id = task.task_id;
+      const due_date = task.due_date;
+      const assignee = task.assignee_user_id;
+      const users = userFarms[farm_id];
+      let assigneeRole;
+
+      if (assignee) {
+        assigneeRole = Object.values(users).find((user) => user.user_id === assignee).role_id;
+      }
+      let pathName;
+
+      if (new Date(due_date) <= new Date() && (user_id === assignee || assigneeRole === 4)) {
```

Comment:
Could you add a comment that role id 4 is worker or use `roleToId` object? I think the variable name should be `assigneeRoleId`.